lib/Command/SetSecretVerificationStatusCommand.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * @author Semih Serhat Karakaya <karakayasemi@itu.edu.tr>
+ *
+ * Two-factor TOTP
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\TwoFactor_Totp\Command;
+
+use InvalidArgumentException;
+use OCA\TwoFactor_Totp\Db\TotpSecretMapper;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\IUserManager;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SetSecretVerificationStatusCommand extends Command {
+
+	/** @var TotpSecretMapper */
+	private $secretMapper;
+
+	/** @var IUserManager */
+	private $userManager;
+
+	public function __construct(
+		TotpSecretMapper $secretMapper,
+		IUserManager $userManager
+	) {
+		parent::__construct();
+		$this->secretMapper = $secretMapper;
+		$this->userManager = $userManager;
+	}
+
+	protected function configure() {
+		$this
+			->setName('twofactor_totp:set-secret-verification-status')
+			->setDescription(
+				'Set secret verification of specified users or all users'
+			)
+			->addArgument(
+				'set-verified',
+				InputArgument::REQUIRED,
+				'Secret verification status to set. (true or false)'
+			)
+			->addOption(
+				'all',
+				null,
+				InputOption::VALUE_NONE,
+				'Will affects all users secrets'
+			)
+			->addOption(
+				'uid',
+				'u',
+				InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+				'The user\'s uid is used. This option can be used as --uid "Alice" --uid "Bob"'
+			);
+	}
+
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return int
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		try {
+			$options = $this->validateInput($input);
+		} catch (InvalidArgumentException $e) {
+			$message = $e->getMessage();
+			$output->writeln("<error>$message</error>");
+			return 1;
+		}
+		$status = $options['status'];
+		if($status === true) {
+			$commandActionString = 'verified';
+		} else {
+			$commandActionString = 'unverified';
+		}
+
+		if ($options['uids'] === 'all') {
+			$this->secretMapper->setAllSecretsVerificationStatus($status);
+			$output->writeln("<info>All the secrets are $commandActionString</info>");
```

Comment:
"All the secrets are true". and so are all conspiration theories ? :-D

Maybe "All the TOTP secrets have been set to true/false" or "set to verified/unverified" would sound better ?

cc @phil-davis

Reply:
The status of all TOTP secrets has been set to verified/unverified